Sources/NIOUDPEchoClient/main.swift

```diff
@@ -126,14 +126,23 @@ defer {
     try! group.syncShutdownGracefully()
 }
 
-let channel = try { () -> Channel in
-    switch connectTarget {
-    case .ip(let host, _, let listeningPort):
-        return try bootstrap.bind(host: host, port: listeningPort).wait()
-    case .unixDomainSocket(_, let listeningPath):
-        return try bootstrap.bind(unixDomainSocketPath: listeningPath).wait()
+
+let channel: Channel
+let connectedMode = true
```

Comment:
Currently this is unconditionally `true`: did you want to add support for both?

Reply:
Yeah, that's what I meant by this in the considerations in the PR description:

> Is the NIOUDPEchoClient considered API? If not, can we rework the argument parsing so we can include a --connect flag?

If it's not considered a breaking change then I can add an option. Adding an option with the existing argument parsing strategy is gonna be pretty clunky.

Reply:
We can rewrite the argument parsing here.

Reply:
Is depending on `swift-argument-parser` a no-go?

Reply:
Added commit:

```
* 4d5652cc6 NIOUDPEchoClient: Connect socket to remote only if --connect is used
```